# Generated by Django 5.0.4 on 2024-11-04 22:58

import django.db.models.deletion
from django.db import migrations, models


class Migration(migrations.Migration):

    dependencies = [
        ('base', '0003_color_size_product_colors_productimage_product_sizes'),
    ]

    operations = [
        migrations.CreateModel(
            name='OrderItem',
            fields=[
                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('session_key', models.CharField(max_length=32)),
                ('quantity', models.IntegerField(default=1)),
                ('ordered', models.BooleanField(default=False)),
                ('product', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, to='base.product')),
            ],
        ),
        migrations.CreateModel(
            name='Order',
            fields=[
                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('session_key', models.CharField(max_length=32)),
                ('ordered', models.BooleanField(default=False)),
                ('being_delivered', models.BooleanField(default=False)),
                ('received', models.BooleanField(default=False)),
                ('ordered_date', models.DateTimeField(null=True)),
                ('remark', models.CharField(blank=True, default='', max_length=255, null=True)),
                ('order_status', models.CharField(blank=True, choices=[('printed', 'printed'), ('confirmed_change_delivery_time', 'confirmed_change_delivery_time'), ('error', 'error'), ('kitchen', 'kitchen'), ('in_delivery', 'in_delivery'), ('delivered', 'delivered')], max_length=50, null=True)),
                ('items', models.ManyToManyField(to='base.orderitem')),
            ],
        ),
    ]
